supervisor/loggers.py

```diff
@@ -387,10 +396,15 @@ def handle_file(logger, filename, fmt, rotating=False, maxbytes=0, backups=0):
         if rotating is False:
             handler = FileHandler(filename)
         else:
-            handler = RotatingFileHandler(filename, 'a', maxbytes, backups)
+            handler = RotatingFileHandler(filename, 'a', maxbytes, backups,
+                                          compress)
 
     handler.setFormat(fmt)
     handler.setLevel(logger.level)
     logger.addHandler(handler)
 
     return logger
+
+def handle_compression(sfn, dfn):
+    with tarfile.open(dfn, "w:gz") as archive:
+        archive.add(sfn, arcname=os.path.basename(sfn))
```

Comment:
Didn't consider that as I'm not 100% familiar with Supervisor's architecture.

Sure a subprocess could easily handle the the compression without blocking the daemon. I can provide such implementation if needed.